src/Gui/MaterialParameterEditor/MaterialParameterEditor.cpp

```diff
@@ -0,0 +1,356 @@
+        std::string description = m.contains( "description" ) ? m["description"] : "";
+
+        m_parametersControlPanel->addVectorInput(
+            m["name"], onVectorParameterChanged, init, dec, description );
+    }
+    else if ( m_showUnspecified ) {
+        m_parametersControlPanel->addVectorInput( key, onVectorParameterChanged, init );
+    }
+}
+
+template <typename T>
+void MaterialParameterEditor::addMatrixParameterWidget( const std::string& key,
+                                                        const T& initial,
+                                                        Ra::Engine::Data::RenderParameters& params,
+                                                        const json& metadata ) {
+    auto onMatrixParameterChanged = [this, &params, &key]( const Ra::Core::MatrixN& value ) {
+        auto mat3 = T( value );
+        params.addParameter( key, mat3 );
+        emit materialParametersModified();
+    };
+
+    if ( metadata.contains( key ) ) {
+        auto m                  = metadata[key];
+        std::string description = m.contains( "description" ) ? m["description"] : "";
+
+        m_parametersControlPanel->addMatrixInput(
+            m["name"], onMatrixParameterChanged, initial, 3, description );
+    }
+    else if ( m_showUnspecified ) {
+        m_parametersControlPanel->addMatrixInput( key, onMatrixParameterChanged, initial );
+    }
+}
+
+void MaterialParameterEditor::setupFromMaterial(
+    std::shared_ptr<Ra::Engine::Data::Material> material ) {
+    auto hasMetadata =
+        std::dynamic_pointer_cast<Ra::Engine::Data::ParameterSetEditionInterface>( material );
+    m_matNameLabel->setText( QString::fromStdString( material->getMaterialName() ) );
+    m_matProperties->clear();
+    for ( auto& prop : material->getPropertyList() ) {
+        m_matProperties->appendPlainText( QString::fromStdString( prop ) );
+    }
+    // move the cursor to the top
+    auto tmpCursor = m_matProperties->textCursor();
+    tmpCursor.movePosition( QTextCursor::Start );
+    m_matProperties->setTextCursor( tmpCursor );
+
+    auto& params  = material->getParameters();
+    auto metadata = ( hasMetadata ) ? hasMetadata->getParametersMetadata() : nlohmann::json {};
+
+    setupFromParameters( params, metadata, material->getInstanceName() );
+
+    m_matInfoGroup->setVisible( true );
+}
+
+void MaterialParameterEditor::setupFromParameters( Engine::Data::RenderParameters& params,
+                                                   const nlohmann::json& constraints,
+                                                   const std::string& name ) {
+
+    m_matInfoGroup->setVisible( false );
+    if ( !name.empty() ) { m_matInstanceNameLabel->setText( QString::fromStdString( name ) ); }
+    else {
+        m_matInstanceNameLabel->setText( "Parameter set" );
+    }
+    // clear the old control panel
+    m_matParamsLayout->removeWidget( m_parametersControlPanel );
+    delete m_parametersControlPanel;
```

Comment:
I hade some weird segfault on windows (in other code base), directly deleting widget, I haven't tested this one but maybe use 
```suggestion
    m_parametersControlPanel->deleteLater();
```
see https://doc.qt.io/qt-6/qobject.html#deleteLater

Reply:
uhm ... not sure deleteLater is the solution. When deleting a widget, all child widgets are first deleted. Can you show me, by private channel, what you're doing in the other code-base ? Do you add custom widgets to the panel ? How are they managed ? I'll verify this on windows ASAP, with the edition demonstrator to see if the crash is due to this delete or to an eventually added externally added widget.I'll also remind in the documentation that the ControlPanel takes ownership of the added widgets.

Reply:
The crashed I had are not due to MaterialParameterEditor but to other similar widget.
If I delete directly the widget, they still receive some event, but maybe here it's not the case.
deleteLater simply ensure there is no current event propagation that can trigger event from a widget that will be deleted.
The following snippet crash on windows if I call delete child->widget() instead of deleteLater
```
        auto vlayout = layout();
        if ( vlayout ) {
            QLayoutItem* child;
            while ( ( child = vlayout->takeAt( 0 ) ) != nullptr ) {
                if ( child->widget() ) {
                    child->widget()->disconnect();
                    child->widget()->deleteLater(); // delete the widget
                }
                delete child; // delete the layout item
            }
        }
        else {
            vlayout = new QVBoxLayout( this );
        }
```

Reply:
I've tested on windows, the MaterialEditionDemo is now operational. I've not meet your problem.